onedal/__init__.py

```diff
@@ -46,7 +46,7 @@
     __all__.append('spmd')
 
 if daal_check_version((2023, 'P', 100)):
-    __all__ += ['basic_statistics', 'linear_model']
+    __all__ += ['basic_statistics', 'linear_model', 'objective_function']
```

Comment:
```suggestion
    __all__ += ['basic_statistics', 'linear_model', 'metrics']
```
I suppose it should be in `metrics` module

Reply:
Metric is the name of function that we use to measure the quality of our predictions, while objective function is something that we use to fit the algorithm (calculate gradient, do optimization steps and etc.) Moreover, all the modules in onedal are already named objective_function so if we use metrics here this would lead to confusion. So I would not agree with this comment

Reply:
The `metric` proposed here is just module name, not a function, like in scikit-learn, that includes `log_loss and other classification and regression metric functions.

We don't have any naming rules here. For example, in sklearnex/onedal we are mostly following scikit-learn like interface, so naming as well.